app/src/main/java/com/example/cachua/musicdemo21/ui/music/MusicActivity.kt

```diff
@@ -0,0 +1,196 @@
+package com.example.cachua.musicdemo21.ui.music.ui.music
+
+import android.Manifest
+import android.content.*
+import android.content.pm.PackageManager
+import android.support.v7.app.AppCompatActivity
+import android.os.Bundle
+import android.os.IBinder
+import android.support.v4.app.ActivityCompat
+import android.support.v4.content.ContextCompat
+import android.support.v7.widget.StaggeredGridLayoutManager
+import android.util.Log
+import android.view.View
+import android.widget.SeekBar
+import android.widget.Toast
+import com.bumptech.glide.Glide
+import com.example.cachua.musicdemo21.R
+import com.example.cachua.musicdemo21.data.model.MusicModel
+import com.example.cachua.musicdemo21.data.repository.MusicRepository
+import com.example.cachua.musicdemo21.data.source.ContentResolverData
+import com.example.cachua.musicdemo21.data.source.MusicLocalDatasource
+import com.example.cachua.musicdemo21.service.MyMusicService
+import com.example.cachua.musicdemo21.ui.music.MusicContract
+import com.example.cachua.musicdemo21.ui.music.MusicPresenter
+import com.example.cachua.musicdemo21.ui.music.adapter.MusicAdapter
+import com.example.cachua.musicdemo21.utils.Constant
+import kotlinx.android.synthetic.main.activity_music.*
+import java.text.SimpleDateFormat
+import java.util.*
+import kotlin.collections.ArrayList
+
+class MusicActivity : AppCompatActivity(), MusicContract.View, MyMusicService.CallBack, View.OnClickListener, MusicAdapter.CallBack {
+
+    private lateinit var mMusicPresenter: MusicPresenter
+    private lateinit var mMusicRepository: MusicRepository
+    private lateinit var mMusicLocalDataSource: MusicLocalDatasource
+    private lateinit var mMusicAdapter: MusicAdapter
+    private lateinit var simpleDateFormat: SimpleDateFormat
+    private lateinit var mMyMusicService: MyMusicService
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_music)
+        initView()
+    }
+
+    private fun initView() {
+        initData()
+        initEvent()
+    }
+
+    private fun initData() {
+        val mContentProvider = ContentResolverData(this)
+        mMusicLocalDataSource = MusicLocalDatasource(mContentProvider)
+        mMusicRepository = MusicRepository(mMusicLocalDataSource)
+        mMusicPresenter = MusicPresenter(mMusicRepository, this)
+        simpleDateFormat = SimpleDateFormat("mm:ss", Locale.US)
+        
+        var intent = Intent(this, MyMusicService::class.java)
+        bindService(intent, mServiceConnection, Context.BIND_AUTO_CREATE)
+        
+        var mArrayMusic= ArrayList<MusicModel>()
+        mMusicAdapter = MusicAdapter(applicationContext, mArrayMusic)
+        recyclerMusic.layoutManager = StaggeredGridLayoutManager(1, StaggeredGridLayoutManager.VERTICAL)
```

Comment:
Không cần truyền context vào adapter, sử dụng apply{} để code gọn gàng hơn. 
```suggestion
         mMusicAdapter = MusicAdapter(emptyList<MusicModel>())
    recyclerMusic?.apply{
        adapter = mMusicAdapter
        layoutManager = StaggeredGridLayoutManager(1, StaggeredGridLayoutManager.VERTICAL)
        setHasFixedSize(true)        
    }
```